### ISSUE TITLE
Typing `import an` should not result in auto-importing of antigravity

### ISSUE THREAD

**Issue by ltfish** [microsoft/pylance-release#4229]
## Environment data

-   Language Server version: 2023.4.10 (pyright d7616109)
-   OS and version: Windows 10 22H2
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.11

## Code Snippet

```python
import an
```

## Repro Steps

1. Ensure the Python and Pylance extensions are installed and enabled.
2. Create a `test.py` file. Save it somewhere.
3. Type `import an`. Wait for the suggestion list to pop up.
4. If `antigravity` is not the top item, keep typing `g` after `import an`.
5. The default browser launches and navigates to https://xkcd.com/353/.

## Expected behavior

Whether `antigravity` is to be imported or not, the default browser should not launch.

## Actual behavior

The default browser launches and navigates to https://xkcd.com/353/.

## Logs

I feel the intended behavior for the Python extension is to not auto-import `antigravity` when it's at the top of the import suggestion pop-up. A blacklist of what modules to not auto-import might suffice. Please let me know if logs are required for diagnosing this issue, and I will collect logs.

**Comment by vibhapathak** [comment]
Is this issue fixed?

**Comment by debonte** [comment]
When I follow your repro steps I'm not seeing the browser launch. `antigravity` does [launch your web browser](https://github.com/python/cpython/blob/main/Lib/antigravity.py#L5) when executed. But Pylance doesn't execute the code that it analyzes. Running my app does cause the browser to launch as expected.

I agree that excluding Easter egg modules like `antigravity` from auto-import makes sense.

Related to https://github.com/microsoft/pylance-release/issues/2535 which is about importing `this`.

**Comment by ltfish** [comment]
It might be caused by other extensions that I installed, but I am not a frequent VS Code user so I didn't install many other extensions.

I will try to reproduce it in a clean install of VS Code.

**Comment by judej** [comment]
Dupe of #2535.

If you are able to provide repro steps for the browser open issue, please reopen this issue.

**Comment by ltfish** [comment]
Managed to get a traceback when `antigravity` was imported by Pylance:

```
File "c:\Users\Fish\.vscode\extensions\ms-python.vscode-pylance-2023.4.10\dist\scripts\scrape_module.py", line 1302, in <module>
    main()
File "c:\Users\Fish\.vscode\extensions\ms-python.vscode-pylance-2023.4.10\dist\scripts\scrape_module.py", line 1291, in main
    state = ScrapeState(module_name, search_path)
File "c:\Users\Fish\.vscode\extensions\ms-python.vscode-pylance-2023.4.10\dist\scripts\scrape_module.py", line 1086, in __init__
    self.module = do_import(self.module_name, search_path)
File "c:\Users\Fish\.vscode\extensions\ms-python.vscode-pylance-2023.4.10\dist\scripts\scrape_module.py", line 1050, in do_import
    return importlib.import_module(module_name)
File "D:\My Program Files\Python311\Lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
File "<frozen importlib._bootstrap>", line 1149, in _find_and_load_unlocked
File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
File "<frozen importlib._bootstrap_external>", line 940, in exec_module
File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
File "D:\My Program Files\Python311\Lib\antigravity.py", line 3, in <module>
    for line in traceback.format_stack():
```

Looks like Pylance does perform import?

**Comment by ltfish** [comment]
So I don't think this issue is an duplicate of #2535 because it has nothing to do with the auto-import feature. The problem seems to be caused by `scrape_module.py` starting to analyze the `antigravity` module when `antigravity` is selected from the suggested import pop-up list. I'll need to debug more to see why `scrape_module.py` is triggered in the first place.

**Comment by debonte** [comment]
Thanks @ltfish, I'm able to force this to happen while debugging Pylance, but I haven't figured out how to really repro it yet.

On my machine, when `PylanceImportResolve.getSourceFilesFromStub` calls `super.getSourceFilesFromStub` is always gets back a list containing `c:\Users\erikd\AppData\Local\Programs\Python\Python311\Lib\antigravity.py` and returns before scraping occurs because that path doesn't pass the "special cases" regex.

**Comment by ltfish** [comment]
> Also, I think it was hovering over the antigrravity module name that triggered this code path rather than adding the import.

That is correct. Sorry for not making it clear in my original repro comment.

**Comment by bschnurr** [comment]
If you right click "go to declaration" does it find a `antigravity.pyi` file?

**Comment by ltfish** [comment]
It does find (and navigate) to an `antigravity.pyi` file at `C:\Users\Fish\.vscode\extensions\ms-python.vscode-pylance-2023.4.10\dist\typeshed-fallback\stdlib\antigravity.pyi`.

**Comment by heejaechang** [comment]
the repro step describes in the issue is not `auto-import` rather it is regular `import completion`

and `scrape_module.py` is to generate `py` over `native module`. so it shouldn't matter for `antigravity`. it probably ran while generating tooltip for `antigravity` since some of data it needs were from `native module` such as `builtin` types. we only run it for every specific builtin/stdlib modules and we run it only once per module per session, so it shouldn't run repeatedly for the same module.

since we never run actual `py` file, I am not sure how we make the module to run without user asking for it. to dig in more, we probably need actual repro steps.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2023.5.11, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023511-3-May-2023-PreRelease)

**Comment by ltfish** [comment]
Thank you for the fix!
